### PR TITLE
Show failed tests on CircleCI layout in a better way

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

I am not sure if this will helps the navigation (much), but at least for the team members to try.

A new step is added to show the (only faiiled) test results, and it looks like
Screenshot 2023-08-31 184247

The `Run tests` step won't fail now (otherwise we can't have the new `Check test results` step), but the new step will (if there is indeed a failed test).

For a job run page, see [here](https://app.circleci.com/pipelines/github/huggingface/transformers/71854/workflows/8151a27f-4617-4781-9d0d-80bcb4847edf/jobs/905042)

WDYT?

### CODE REVIEW

File: .circleci/create_circleci_config.py [hunk 1]
```diff
@@ -223,18 +223,28 @@ def to_dict(self):
...
-            checkout_doctest_command += 'cat reports/tests_pr_documentation_tests/summary_short.txt; exit -1; '
-            checkout_doctest_command += 'elif [ -s reports/tests_pr_documentation_tests/stats.txt ]; then echo "All tests pass!"; '
-            checkout_doctest_command += 'elif [ -f 124.txt ]; then echo "doctest timeout!"; else echo "other fatal error)"; exit -1; fi;'
-            steps.append({"run": {"name": "Check doctest results", "command": checkout_doctest_command}})
+            check_test_command += 'elif [ -f 124.txt ]; then echo "doctest timeout!"; '
+
+        check_test_command += 'else echo "other fatal error"; exit -1; fi;'
+        check_test_command += 'echo ""; '
+
+        steps.append({"run": {"name": "Check test results", "command": check_test_command}})
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply .circleci/create_circleci_config.py:247]: So far I put `failures_short.tx`. We can use `failures_long.tx` if you prefer.

**Review by ArthurZucker — COMMENTED** [review]
I'm not 100% sure there is a lot of difference between this and looking at the artifacts. Also if we do this, we could also remove the call to | tee following #23027. 
This way you can either have a nice looking output on the page or go and check the artifacts.

**Comment by ydshieh** [comment]
No problem to close if you maintainers don't feel it makes navigation (even slightly)  better/faster.

**Comment by ArthurZucker** [comment]
I'm actually in favor if we also remove tee 😉

**Review by amyeroberts — APPROVED** [review]
The output is so much better - thank you!

**Comment by ydshieh** [comment]
@amyeroberts  BTW, the `tests_output.txt` will be removed to make @ArthurZucker  happier. But want to have your confirmation too.

**Comment by ydshieh** [comment]
No more tea Sir, sorry.

**Review by ArthurZucker — COMMENTED** [review]
Arf, it seems that we still don't have the nice output I was hoping for (with colors and more readable errors). Do you know why / could you try aiming for this? 

ex:
image

**Comment by ydshieh** [comment]
I need a run (page) that already has such colors and format.

Personally, I don't feel it's necessary to have it - as we (mostly if not always) care the failing ones. The new step already give ONLY the failed tests without the passing and skipped one - so the color doesn't matter (at least for me).

I would move on to merge this PR as you and amy is OK with the changes. For the color thing, if you can provide a run page, I could probably give it a try in the future.

**Comment by ArthurZucker** [comment]
Sure 😉 feel free to merge!

**Comment by ydshieh** [comment]
We do have color, but it only show the failing ones. See [here](https://app.circleci.com/pipelines/github/huggingface/transformers/72192/workflows/c446000b-bf65-4531-a4ea-13934ef594ec/jobs/910150).

On current main, no RED color at all.

Screenshot 2023-09-05 152009

**Comment by amyeroberts** [comment]
Happy for `tests_output.txt` to go - all the info can be found in the other artefacts

**Comment by ArthurZucker** [comment]
Perfect!

**Pull request merged**